### DISCUSSION TITLE
Update email templates using CLI

### DISCUSSION THREAD

**Discussion by luizcieslak** [supabase/supabase#8165] [Feature Requests]
Hi! 

I wanted to ask if we can have a way to update these email templates using the `supabase-cli`?

![image](https://user-images.githubusercontent.com/14146176/183668456-6c9e9ba3-4b36-48a1-922a-9510e8e0149b.png)

The use case is for developers already using email frameworks like [mjml](mjml.io) to build structured email layouts without the hassle. 

Right now the only way of updating them is to build the HTML string locally and manually copy and paste it in the Supabase dashboard. 😢 

Checking #2831, it seems you have plans to implement a template editor in a future date. However, it would be awesome to update them outside the dashboard as well.

The usage could be something like

```bash
supabase email update confirm_password "html string here"
# or
supabase email update confirm_password --file ./layouts/confirm_password.html
```

This would also enable developers to hook these commands into task runners like GitHub Actions, ensuring the templates are up-to-date after a local change.

TY!

**Comment by cdimitroulas** [comment]
This would be a great feature to have in general, even if you aren't using an email framework like `mjml`. If you have any need to customize email templates then you will need to add the HTML to your codebase in order to develop locally (since for local development you need to use `config.toml` to load the custom email templates rather than doing it via the Supabase studio).

Being able to automate that any changes to those email templates in the codebase are automatically deployed to the Supabase environments using CI/CD using the Supabase CLI would be great as otherwise it adds another manual step to deployments in the case of email template changes (and manual steps for deployments are obviously something that can easily be forgotten 😅)

**Comment by lucasyvas** [comment]
By extension, having this capability via API seems like a must. More generally, you should ideally be able to programmatically update email templates.

Firebase is very self-serve friendly in this respect - you can set up literally anything to do with a project via their API.

**Comment by vickkhera** [comment]
The CLI/API would also need to set the message subject line.

**Comment by swizzmagik** [comment]
Any update on this feature? Seems like it should be straightforward?

**Comment by edzzn** [comment]
this exists!

first:

https://supabase.com/docs/guides/local-development/customizing-email-templates

to push:


`supabase config push` works

https://supabase.com/docs/reference/cli/supabase-config-push

**Reply by vickkhera** [reply]
The configuration setting feature in the recently released supabase CLI v2 is a good step forward. However, they provide no solution to manage multiple deployments, such as staging and production. It always wants to use the `config.toml` file, which for me is tuned for local development.
